sound/soc/sof/topology.c

```diff
@@ -2230,6 +2275,25 @@ static int sof_widget_ready(struct snd_soc_component *scomp, int index,
 		strnlen(tw->sname, SNDRV_CTL_ELEM_ID_NAME_MAXLEN) > 0
 			? tw->sname : "none");
 
+	ret = sof_parse_tokens(scomp, &comp, core_tokens,
+			       ARRAY_SIZE(core_tokens), tw->priv.array,
+			       le32_to_cpu(tw->priv.size));
+	if (ret != 0) {
+		dev_err(scomp->dev, "error: parse dai.comp tokens failed %d\n",
+			ret);
+		return ret;
+	}
+
+	if ((int)comp.core < 0) {
```

Comment:
Hmm, this looks iffy. If comp.core is not signed, I'd have a separate more explicit safe conversion from the unsigned field to the integer value (i.e. must be under certain max value).

Reply:
it actually just checks, if it has been modified from the initialisation value, I can in fact even just check `== -ENOENT`. Well, if `comp.core` becomes 16 bits at some point, then this will fail, yes... Is this what you're concerned about? I think the safest would be just to initialise it to 0 then and power it up unconditionally.